adagrams/game.py

```diff
@@ -1,13 +1,132 @@
 from random import randint
 
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+HAND_SIZE = 10
+PILE_SIZE = len(LETTER_POOL)
+
+def make_new_draw_pile():
+    draw_pile = []
+    for letter, count in LETTER_POOL.items():
+        [draw_pile.append(letter)] * count
+    return draw_pile
+
 def draw_letters():
-    pass
+    new_pile = make_new_draw_pile()
+    new_pile_len = PILE_SIZE
+    hand = []
+
+    for _ in range(HAND_SIZE):
+        hand.append(new_pile.pop(randint(0, new_pile_len - 1)))
+        new_pile_len -= 1
+
+    return hand
+
+def counting(count_what, in_what):
+    total_count = 0
+    for i in in_what:
+        if i == count_what:
+            total_count += 1
+
+    return total_count
 
 def uses_available_letters(word, letter_bank):
-    pass
+    if not word:
+        return False
+    
+    for letter in word:
+        if letter.upper() in letter_bank and counting(letter.upper(), 
+                    word.upper()) <= counting(letter.upper(), letter_bank):
+            continue
+        else:
+            return False
+        
+    return True
 
 def score_word(word):
-    pass
+    SCORE_CHART = { 
+        'A': 1, 'B': 3, 'C': 3, 'D': 2, 'E': 1, 'F': 4, 'G': 2, 'H': 4, 
+        'I': 1, 'J': 8, 'K': 5, 'L': 1, 'M': 3, 'N': 1, 'O': 1, 'P': 3,
+        'Q': 10, 'R': 1, 'S': 1, 'T': 1, 'U': 1, 'V': 4, 'W': 4, 'X': 8,
+        'Y': 4, 'Z': 10
+                    }
```

Comment:
👍 Revising the layout of the score significantly simplifies the scoring logic below.